File: android-app/app/src/main/java/com/example/campusconnect/phase1/model/Event.java
package com.example.campusconnect.phase1.model;

public class Event {
    public String title;
    public String description;
    public double latitude;
    public double longitude;
    public String event_time;
    public int created_by;

    public Event(String title, String description, String event_time,
                 double latitude, double longitude, int created_by) {
        this.title = title;
        this.description = description;
        this.event_time = event_time;
        this.latitude = latitude;
        this.longitude = longitude;
        this.created_by = created_by;
    }
}

File: android-app/app/src/main/java/com/example/campusconnect/model/Event.kt
package com.example.campusconnect.model

data class Event(

    val id: Int,

    // 🔹 BASIC
    val title: String,
    val description: String? = null,

    // 🔹 LOCATION
    val latitude: Double,
    val longitude: Double,

    // UI positioning (temporary for map)
    val xRatio: Float = 0.5f,
    val yRatio: Float = 0.5f,

    // 🔹 TIME
    val date: String = "",              // added separately
    val startTime: String = "",
    val endTime: String? = null,

    // 🔹 ORGANIZATION
    val createdBy: Int = 0,
    val clubName: String = "",

    // 🔹 MEDIA
    val isPoster: Boolean = false,
    val posterUrl: String? = null,

    // 🔹 CLASSIFICATION
    val category: String = "",

    // 🔹 VISIBILITY
    val visibilityType: String = "",
    val visibilityValue: String = "",

    // 🔹 REGISTRATION
    val registrationRequired: Boolean = false,
    val registrationLink: String = "",
    val inAppRegistration: Boolean = false,

    // 🔹 EXTRA
    val venue: String = ""
)